test/test_field.jl

```diff
@@ -451,4 +450,65 @@ end
+
+            k_top = (FieldType == ZFaceField && ZTopology == Bounded) ? Nz+1 : Nz 
+
+            c = FieldType(grid)
+            set!(c, (x, y, z) -> rand())
+
+            CUDA.@allowscalar begin
+                # First test that the regular view is correct
+                cv = view(c, :, :, 1+1:k_top-1)
+                # cv = view(c, :, :, 2:7) for a ZFaceField with bounded topology in z else cv = view(c, :, :, 2:6).
+                @test size(cv) == (1, 1, k_top-2)
+                @test size(parent(cv)) == (1+2Hx, 1+2Hy, k_top-2)
+                @test all(cv[1, 1, i] == c[1, 1, i] for i in 1+1:k_top-1)
+
+                # Now test the views of views
+                cvv = view(cv, :, :, 1+2:k_top-2)
+                # cvv = view(cv, :, :, 3:6) for a ZFaceField with bounded topology in z else cvv = view(cv, :, :, 3:5).
+                @test size(cvv) == (1, 1, k_top-4)
+                @test size(parent(cvv)) == (1+2Hx, 1+2Hy, k_top-4)
+                @test all(cvv[1, 1, i] == cv[1, 1, i] for i in 1+2:k_top-2)
+
+                cvvv = view(cvv, :, :, 1+3:k_top-3)
+                # cvvv = view(cvv, :, :, 4:5) for a ZFaceField with bounded topology in z else cvvv = view(cvv, :, :, 4:4).
+                @test size(cvvv) == (1, 1, k_top-6)
+                @test size(parent(cvvv)) == (1+2Hx, 1+2Hy, k_top-6)
+                @test all(cvvv[1, 1, i] == cvv[1, 1, i] for i in 1+3:k_top-3)
+
+                @test_throws ArgumentError view(cv, :, :, 1)
+                @test_throws ArgumentError view(cv, :, :, k_top)
+                # @test_throws ArgumentError view(cv, :, :, 8) for a ZFaceField with bounded topology in z else @test_throws ArgumentError view(cv, :, :, 7).
+                @test_throws ArgumentError view(cvv, :, :, 1:1+1)
+                @test_throws ArgumentError view(cvv, :, :, k_top-1:k_top)
+                # @test_throws ArgumentError view(cvv, :, :, 6:7) for a ZFaceField with bounded topology in z else @test_throws ArgumentError view(cvv, :, :, 5:6).
+                @test_throws ArgumentError view(cvvv, :, :, 1:1+2)
+                @test_throws ArgumentError view(cvvv, :, :, k_top-2:k_top)
+                # @test_throws ArgumentError view(cvvv, :, :, 6:8) for a ZFaceField with bounded topology in z else @test_throws ArgumentError view(cvvv, :, :, 5:7).
+
+                @test_throws BoundsError cv[:, :, 1]
+                @test_throws BoundsError cv[:, :, k_top]
+                # @test_throws BoundsError cv[:, :, 8] for a ZFaceField with bounded topology in z else @test_throws BoundsError cv[:, :, 7].
+                @test_throws BoundsError cvv[:, :, 1:1+1]
+                @test_throws BoundsError cvv[:, :, k_top-1:k_top]
+                # @test_throws BoundsError cvv[:, :, 7:8] for a ZFaceField with bounded topology in z else @test_throws BoundsError cvv[:, :, 6:7].
+                @test_throws BoundsError cvvv[:, :, 1:1+2]
+                @test_throws BoundsError cvvv[:, :, k_top-2:k_top]

```

Comment:
should this use `interior`? We don't ever directly index fields in the user interface

```suggestion
                @test_throws BoundsError cvvv[:, :, k_top-2:k_top]

```

Reply:
Unfortunately, not, unless we also extend the interior index range like we did with the parent index range. Consider the following MWE:

```julia
julia> using Oceananigans

julia> Nx, Ny, Nz = 1, 1, 7;

julia> grid = RectilinearGrid(size=(Nx, Ny, Nz), x=(0, 1), y=(0, 1), z=(0, 1), topology = (Periodic, Periodic, Bounded));

julia> k_top = Nz + 1;

julia> c = ZFaceField(grid)
1×1×8 Field{Center, Center, Face} on RectilinearGrid on CPU
├── grid: 1×1×7 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 1×1×3 halo
├── boundary conditions: FieldBoundaryConditions
│   └── west: Periodic, east: Periodic, south: Periodic, north: Periodic, bottom: Nothing, top: Nothing, immersed: ZeroFlux
└── data: 3×3×14 OffsetArray(::Array{Float64, 3}, 0:2, 0:2, -2:11) with eltype Float64 with indices 0:2×0:2×-2:11
    └── max=0.0, min=0.0, mean=0.0

julia> set!(c, (x, y, z) -> rand())
1×1×8 Field{Center, Center, Face} on RectilinearGrid on CPU
├── grid: 1×1×7 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 1×1×3 halo
├── boundary conditions: FieldBoundaryConditions
│   └── west: Periodic, east: Periodic, south: Periodic, north: Periodic, bottom: Nothing, top: Nothing, immersed: ZeroFlux
└── data: 3×3×14 OffsetArray(::Array{Float64, 3}, 0:2, 0:2, -2:11) with eltype Float64 with indices 0:2×0:2×-2:11
    └── max=0.72276, min=0.0939695, mean=0.428961

julia> cv = view(c, :, :, 1+1:k_top-1)
1×1×6 Field{Center, Center, Face} on RectilinearGrid on CPU
├── grid: 1×1×7 RectilinearGrid{Float64, Periodic, Periodic, Bounded} on CPU with 1×1×3 halo
├── boundary conditions: FieldBoundaryConditions
│   └── west: Periodic, east: Periodic, south: Periodic, north: Periodic, bottom: Nothing, top: Nothing, immersed: ZeroFlux
├── indices: (:, :, 2:7)
└── data: 3×3×6 OffsetArray(view(::Array{Float64, 3}, :, :, 5:10), 0:2, 0:2, 2:7) with eltype Float64 with indices 0:2×0:2×2:7
    └── max=0.72276, min=0.0939695, mean=0.503409

julia> interior(cv)
1×1×6 view(::Array{Float64, 3}, 2:2, 2:2, 5:10) with eltype Float64:
[:, :, 1] =
 0.6491566506900277

[:, :, 2] =
 0.5118574586205709

[:, :, 3] =
 0.6214555441666034

[:, :, 4] =
 0.09396949649555086

[:, :, 5] =
 0.7227599953730516

[:, :, 6] =
 0.4212556767008805
```
Currently, the interior indices start from 1 and extend up to the size of the interior of the field.

Reply:
I guess I'm not totally sure what the test is contributing